dashboard/src/components/util/Query.js

```diff
@@ -0,0 +1,61 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { ApolloError } from "apollo-client";
+import { Query as BaseQuery } from "react-apollo";
+
+import QueryAbortedError from "/errors/QueryAbortedError";
+
+class Query extends React.PureComponent {
+  static propTypes = {
+    onError: PropTypes.func.isRequired,
+    children: PropTypes.func.isRequired,
+  };
+
+  static defaultProps = {
+    onError(error) {
+      throw error;
+    },
+  };
+
+  componentDidMount() {
+    const { onError } = this.props;
+    this.querySubscription = this.queryRef.current.queryObservable.subscribe({
+      next(result) {
+        if (result.errors && result.errors.length > 0) {
+          throw new ApolloError({ graphQLErrors: result.errors });
```

Comment:
This would be the case where the GraphQL endpoint returns an `errors` field on the response. This will cause them to be globally thrown (as we discussed last week)